### DISCUSSION TITLE
Automation examples

### DISCUSSION THREAD

**Discussion by agav85** [ekutner/home-connect-hass#189] [Community Support]
Hello! Thank you very much for your work and this integration. Could you please share the examples of automations, utilizing services provided by this integration. This could be very helpful. Thanks in advance and have a nice day!

**Comment by bengutt** [comment]
I've managed to get this working. I've got a cheap electric rate at 2am, do I want to be able to press a button to set the dishwasher to come on at that time. The first two actions are probably not needed as I could probably do them in the service call.

If you want to use this for a different time, swap out the 7260 to your required time. 7260 is the number of seconds after midnight of 02:01 am.

alias: Start the dish washer at 2am
description: ""
trigger:
  - device_id: 85856a19719eb35c70fd9006755f0770
    domain: zha
    platform: device
    type: remote_button_short_press
    subtype: remote_button_short_press
condition: []
action:
  - device_id: 1884cdf5810fe7d7ee77de617a62b3e1
    domain: select
    entity_id: select.011090527028004609_programs
    type: select_option
    option: Dishcare.Dishwasher.Program.Eco50
  - type: turn_on
    device_id: 1884cdf5810fe7d7ee77de617a62b3e1
    entity_id: switch.011090527028004609_dishcare_dishwasher_option_ecodry
    domain: switch
  - service: home_connect_alt.start_program
    data:
      validate: true
      device_id: 1884cdf5810fe7d7ee77de617a62b3e1
      program_key: Dishcare.Dishwasher.Program.Eco50
      options: |-
        [{
          "key":"BSH.Common.Option.StartInRelative",
          "value":{{ ((86400 - (timedelta(hours = now().hour, minutes = now().minute, seconds = now().second).total_seconds()) + 7260) % 86400 ) | int }}
        }]
mode: single

**Reply by agav85** [reply]
Thank you very much for the provided example. It is absolutely clear. But I have a related question: I see that you also enable Extra Dry option in the example; In addition to that option my dishwasher supports Hygiene Plus and Extra Rinse. But I do not have the switch for Extra rinse available as entity.

**Reply by Steph6431** [reply]
I just installed the integration, was searching for an example and found this post.
So I tried to create an automation with this code but got the following error message when I tried to save it:

Fehler beim Parsen von YAML: can not read a block mapping entry; a multiline key may not be an implicit key (Zeile: 30, Spalte: 5)

Sorry, if this is a stupid question but I'm not very experienced with home assistant so far. ;-)

**Reply by bengutt** [reply]
> I just installed the integration, was searching for an example and found this post. So I tried to create an automation with this code but got the following error message when I tried to save it:
> 
> Fehler beim Parsen von YAML: can not read a block mapping entry; a multiline key may not be an implicit key (Zeile: 30, Spalte: 5)
> 
> Sorry, if this is a stupid question but I'm not very experienced with home assistant so far. ;-)

This is my current version of it, things have changed since I added the original example:
`alias: Kitchen:Start the dish washer at 2am
description: ""
mode: single
triggers:
  - device_id: 85856a19719eb35c70fd9006755f0770
    domain: zha
    type: remote_button_short_press
    subtype: remote_button_short_press
    trigger: device
conditions: []
actions:
  - type: turn_on
    device_id: 1884cdf5810fe7d7ee77de617a62b3e1
    entity_id: 2ae63b7876a99a7e9146d6bb7e6fa893
    domain: switch
  - type: turn_on
    device_id: 1884cdf5810fe7d7ee77de617a62b3e1
    entity_id: switch.011090527028004609_dishcare_dishwasher_option_ecodry
    domain: switch
  - data:
      validate: false
      device_id: 1884cdf5810fe7d7ee77de617a62b3e1
      program_key: Dishcare.Dishwasher.Program.Eco50
      options: |-
        [{
          "key":"BSH.Common.Option.StartInRelative",
          "value":{{ ((86400 - (timedelta(hours = now().hour, minutes = now().minute, seconds = now().second).total_seconds()) + 7260) % 86400 ) | int }}
        }]
    action: home_connect_alt.start_program
`

**Reply by Steph6431** [reply]
Thanks!
I think I'm struggeling with he correct yaml-structure.
Could you please also post a screen shot of this code?

**Reply by HedpE-** [reply]
This is done in a complex way that could be so much simplified, why not trigger at 2am and do whatever you want without the complexity of time calculations? HA does that for you, you just have to trigger @2am...
